scripts/gen-live-site/main.go

```diff
@@ -0,0 +1,271 @@
+		case fs.ModeSymlink:
+			// We treat .js and .css files as normal files and make a
+			// copy, renaming the filename. This removes the symlink and
+			// duplicates the file. Otherwise the symlink is maintained as
+			// in in the dest tree.
+			if ext == ".js" || ext == ".css" {
+				break
+			}
+
+			// Would be nice to be able to ReadLink through a fs.FS:
+			// https://github.com/golang/go/issues/49580
+			l, err := os.Readlink(srcfile)
+			if err != nil {
+				return err
+			}
+			// Create the symlink
+			return os.Symlink(l, destfile)
+
+		case 0: // normal file
+			// handled below
+		default:
+			//nolint:goerr113 // dynamic errors in package main is ok
+			return fmt.Errorf("unknown file type: %s: %s", mode, srcfile)
+		}
+
+		switch ext {
+		case ".html":
+			skippedFiles = append(skippedFiles, filename)
+			return nil
+		case ".js", ".css":
+			shortSha, err := hashFile(srcfile)
+			if err != nil {
+				return err
+			}
+			basename := strings.TrimSuffix(filepath.Base(filename), ext)
+			target := basename + "." + shortSha + ext
+			destfile = filepath.Join(filepath.Dir(destfile), target)
+			if _, ok := renamedFiles[filename]; ok {
+				//nolint:goerr113 // dynamic errors in package main is ok
+				return fmt.Errorf("duplicate filename: %s", srcfile)
+			}
+			renamedFiles[filename] = target
+		}
+		return copyFile(srcfile, destfile)
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	return skippedFiles, renamedFiles, nil
+}
+
+func copyHTMLFiles(src, dest, domain string, skippedFiles []string, renamedFiles map[string]string) error {
+	for _, filename := range skippedFiles {
+		in, out, err := openInOut(filepath.Join(src, filename), filepath.Join(dest, filename))
+		if err != nil {
+			return err
+		}
+		defer in.Close() //nolint:errcheck // don't care about close failing on read-only files
+		err = updateHTMLFile(out, in, filename, domain, renamedFiles)
+		if err != nil {
+			out.Close() //nolint:errcheck,gosec // we're returning the more important error
+			return err
+		}
+		if err := out.Close(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// hashFile returns a short hash of the contents of filename. The short hash is
+// 32 bits, or 8 chars[0-9a-f] and with 100 file changes in a year (cache
+// expiry is one year) has a collision probability of less than 0.0000000005%.
+func hashFile(filename string) (string, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close() //nolint:errcheck // don't care about close failing on read-only files
+	h := sha256.New()
+	if _, err := io.Copy(h, f); err != nil {
+		return "", err
+	}
+	sha := h.Sum(nil)
+	return hex.EncodeToString(sha[:4]), nil
+}
+
+func openInOut(src, dest string) (io.ReadCloser, io.WriteCloser, error) {
+	in, err := os.Open(src)
+	if err != nil {
+		return nil, nil, err
+	}
+	out, err := os.Create(dest)
+	if err != nil {
+		in.Close() //nolint:errcheck,gosec // we're returning the more important error
```

Comment:
I used `_ = in.Close` - removed the linter warning, is shorter and almost as obvious.
Same below.
Up to you if you prefer to keep.

Reply:
I like to be explicit when I'm silencing the linter. I dont like to alter my program just to make the linter happy, and when I do the linter thing I get to put my comment on it as to why.

Reply:
Fair enough, I will update this in `evy serve` PR